src/special_arrays/special_arrays.jl

```diff
@@ -29,6 +29,18 @@ const ImputingMatrix{T, U} = Union{PreImputingMatrix{T, U}, PostImputingMatrix{T
 const PreImputingDense = Dense{T, <: PreImputingMatrix} where T
 const PostImputingDense = Dense{T, <: PostImputingMatrix} where T
 
+# so we error on integers with missings
+Flux.onecold(y::MaybeHotArray{Maybe{T}}, labels = 1:size(y, 1)) where T<:Integer = 
+    throw(ArgumentError("MaybeHotArray{Union{T, Missing}} where T <:Integer can't produce onecold encoding, use maybecold instead."))
+# but we don't error on maybehot which has only integers
+
+y = maybehotbatch([1, missing, 3], 1:10)
+function maybecold(y::AbstractArray, labels = 1:size(y, 1))
```

Comment:
Perhaps specialize to `maybecold(X::MaybeHotMatrix{<:Maybe{Integer}}, labels = 1:size(X, 1))` (for full maybe hot matrices we can fallback to `Flux.onecold`.

I am trying to use capital letters for matrices, `X` for maybe hot matrices in the code

Reply:
Ok, I'll rename the variables.